src/persist-client/src/internal/compact.rs

```diff
@@ -254,14 +275,233 @@ impl Compactor {
         // Compaction is cpu intensive, so be polite and spawn it on the
         // CPU heavy runtime.
         let compact_span = debug_span!("compact::consolidate");
-        cpu_heavy_runtime
+        parts_cpu_heavy_runtime
             .spawn_named(
                 || "persist::compact::consolidate",
                 compact_blocking.instrument(compact_span),
             )
             .await?
     }
 
+    /// WIP: grab the first run from each batch, then the second, etc.
+    fn order_runs<T, D>(req: &CompactReq<T>) -> Vec<(&HollowBatch<T>, &[PartialBatchKey])>
```

Comment:
it feels just the tiniest bit awkward to me for the first element of this tuple to be "HollowBatch except ignore the runs", but it _is_ nice that we can just return refs and not copy anything...

probably fine as-is, just wanted to point this out

Reply:
updated to Description rather than batch since the rest wasn't needed. happy to turn these into named structs if that's more readable too!